### PR TITLE
Deflake integration test caused by concurrent global git config updates

### PR DESCRIPTION
**Pull request by chaodaiG**
Integration test has been pretty flaky, the cause very unlikely was fake git server concurrently set global cookiefile

/cc @listx

### CODE REVIEW

**Comment by chaodaiG** [comment]
Full error message:
```
{"branch":"master","component":"sub","error":"failed to clone repo for \"https://fakegitserver.default/repo/repo3\": unable to configure http.cookiefile: error configuring [--global http.cookiefile /etc/cookies/cookies]: exit status 255 error: could not lock config file /root/.gitconfig: File exists\n","file":"k8s.io/test-infra/prow/pubsub/subscriber/subscriber.go:236","func":"k8s.io/test-infra/prow/pubsub/subscriber.(*presubmitJobHandler).getProwJobSpec","level":"info","msg":"Failed to get presubmits","org":"https://fakegitserver.default/repo","orgRepo":"https://fakegitserver.default/repo/repo3","repo":"repo3","severity":"info","time":"2022-08-26T19:58:27Z"}
```

from https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/27277/pull-test-infra-integration/1563251768412868608

File: prow/test/integration/cmd/fakegitserver/main.go [hunk 1]
```diff
@@ -91,12 +95,12 @@ func main() {
 
 	// Serve HTTPS traffic.
 	if o.cert != "" && o.key != "" {
-		serverHttps := &http.Server{
```
Comment (chaodaiG) [inline prow/test/integration/cmd/fakegitserver/main.go:94]: go-lint suggested this change

**Comment by listx** [comment]
/lgtm
/hold

Thanks Chao!

**Pull request merged**